kitsune/dashboards/readouts.py

```diff
@@ -74,6 +74,21 @@ def get_visits_subquery(document=OuterRef("pk"), period=LAST_30_DAYS):
     )
 
 
+def get_visible_or_no_translation_condition(user, locale):
+    """
+    Returns a queryset expression for Document queries that only includes documents
+    that either do not have a translation for the given locale, or have a translation
+    that is visible to the given user.
+    """
+    return ~Q(translations__locale=locale) | Exists(
```

Comment:
nit: I am wondering if it would be clearer here to create a couple of `Subquery` and combine them into another one.
Eg `get_visible_subquery`, `get_no_translation_subquery and combine those in the filter below